src/modules/SharedModule/components/ModeContext/ModeContext.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, useState } from 'react'
+
+     export const ModeContext = createContext(null);
+
+     export default function ModeContextProvider (props:any) {
+
+       let [mode, setMode] = useState('')
```

Comment:
A good practice is to encapsulate the actions handlers in the context. Currently, we are providing the setter functions and this is not a good practice, as it means that I can pass anything to the setter function.

A better approach is to make something like this:
```
const togglemode = (mode: "dark" | "light")=>{
setMode(mode);
}; 
```
and since we are only having two states only (dark and light) we can make it a boolean instead.

and then pass this function instead of the setter function.